### PR TITLE
[CELEBORN-1799][CIP-14] Add celebornConf to cppClient

### PR DESCRIPTION
**Pull request by HolyLow**
### What changes were proposed in this pull request?
Add CelebornConf to cppClient.

### Why are the changes needed?
The CelebornConf will be used as configuration module in cppClient.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Compilation and UTs.

### CODE REVIEW

**Comment by HolyLow** [comment]
@FMX Could you kindly review this PR? Thanks a lot.

Any suggestion is welcome.

File: cpp/celeborn/conf/CelebornConf.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  }
+  CELEBORN_USER_FAIL("Invalid duration {}", str);
+}
+
+} // namespace
+
+const std::unordered_map<std::string, folly::Optional<std::string>>
+    CelebornConf::kDefaultProperties = {
+        STR_PROP(kRpcLookupTimeout, "30s"),
+        STR_PROP(kClientRpcGetReducerFileGroupRpcAskTimeout, "240s"),
```
Comment (FMX) [inline cpp/celeborn/conf/CelebornConf.cpp:132]: The default value of this config is 30s. We should align the default values.

Comment (HolyLow) [reply cpp/celeborn/conf/CelebornConf.cpp]: Modified as required.

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**